### PR TITLE
allow subscription-only requests for registry list operation

### PR DESCRIPTION
**Pull request by MilesHolland**
# Description

Change the registry list SDK operation to include a `scope` input. This behaves similarly to the same input in for the workspace list operation, in that when it is set to `subscription`, it causes the function to leverage a subscription-only endpoint, instead of the default, subscription and resource group-based endpoint.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/CHANGELOG.md [hunk 1]
```diff
@@ -37,6 +37,7 @@
  - Added a `show_progress` parameter to MLClient for enable/disable progress bars of long running operations.
  - Support `month_days` in `RecurrencePattern` when using `RecurrenceSchedule`.
  - Support `ml_client.schedules.list` with `list_view_type`, default to `ENABLED_ONLY`.
+ - Registry list operation now accepts scope value to allow subscription-only based requests.
```
Comment (needuv) [inline sdk/ml/azure-ai-ml/CHANGELOG.md:40]: this looks like it's in the wrong CHANGELOG section

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_registry_operations.py [hunk 2]
```diff
@@ -52,14 +52,19 @@ def __init__(
         self._init_kwargs = kwargs
 
     #@ monitor_with_activity(logger, "Registry.List", ActivityType.PUBLICAPI)
-    def list(self) -> Iterable[Registry]:
+    def list(self, *, scope: str = "resource_group") -> Iterable[Registry]:
```
Comment (needuv) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_registry_operations.py:55]: consider making this a constant

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_registry_operations.py [hunk 3]
```diff
@@ -52,14 +52,19 @@ def __init__(
...
-        resource group.
+        resource group or subscription.
 
+        :param scope: scope of the listing, "resource_group" or "subscription", defaults to "resource_group"
+        :type scope: str, optional
         :return: An iterator like instance of Registry objects
         :rtype: ~azure.core.paging.ItemPaged[Registry]
         """
-
+        if scope == "subscription":
```
Comment (needuv) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_registry_operations.py:64]: we need to guard against users giving us `subscription` in any casing arrangement. Also, consider making "subscription" a constant
```suggestion
        if scope.lower() == "subscription":
```

File: sdk/ml/azure-ai-ml/CHANGELOG.md (hunk 1 above)
Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/CHANGELOG.md:40]: Moved to features added.

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_registry_operations.py (hunk 2 above)
Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_registry_operations.py:55]: Done

File: sdk/ml/azure-ai-ml/azure/ai/ml/constants/_common.py [hunk 4]
```diff
@@ -562,3 +562,8 @@ class RollingRate:
     DAY = "day"
     HOUR = "hour"
     MINUTE = "minute"
+
+
+class Scope:
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/constants/_common.py:567]: Thx for creating constants for this, can you update worspace_operations to ues these constants too?

Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/azure/ai/ml/constants/_common.py:567]: Done, the only operation that needs it is list, much like registries.

**Pull request merged**